pkgs/development/python-modules/osrparse/default.nix

```diff
@@ -0,0 +1,38 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hypothesis,
+  pytestCheckHook,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "osrparse";
+  version = "7.0.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "kszlim";
+    repo = "osu-replay-parser";
+    tag = "v${version}";
+    hash = "sha256-MBE4z1SDkr0YA5ommF+WZyR2N67Y1/xmDhxrTrUhQJk=";
+  };
+
+  build-system = [ setuptools ];
+
+  optional-dependencies = {
+    dev = [ hypothesis ];
+  };
+
+  nativeCheckInputs = [ pytestCheckHook ] ++ optional-dependencies.dev;
```

Comment:
optional-dep's shouldn't effect the eval/build. If its needed for the check, just add it to the check, please